include/LeMonADE/updater/UpdaterSimpleConnection.h

```diff
@@ -0,0 +1,176 @@
+ * @tparam MoveType name of the specialized move.
+ */
+
+template<class IngredientsType,class MoveType, class ConnectionMoveType>
+class UpdaterSimpleConnection:public AbstractUpdater
+{
+
+public:
+  /**
+   * @brief Standard Constructor initialized with ref to Ingredients and MCS per cycle
+   *
+   * @param ing a reference to the IngredientsType - mainly the system
+   * @param steps MCS per cycle to performed by execute()
+   */
+  UpdaterSimpleConnection(IngredientsType& ing,uint32_t steps = 1 )
+  :ingredients(ing),nsteps(steps){}
+  
+ 
+  /**
+   * @brief This checks all used Feature and applies all Feature if all conditions are met.
+   *
+   * @details This function runs over \a steps MCS and performs the moves.
+   * It setting the age of the system and prints a simple simple simulation speed
+   * in the number of attempted monomer moves per s (tried and performed monomer moves/s).
+   *
+   * @return True if function are done.
+   */
+  bool execute();
+
+  /**
+   * @brief This function is called \a once in the beginning of the TaskManager.
+   *
+   * @details It´s a virtual function for inheritance.
+   * Use this function for initializing tasks (e.g. init SDL)
+   *
+   **/
+  virtual void initialize();
+
+  /**
+   * @brief This function is called \a once in the end of the TaskManager.
+   *
+   * @details It´s a virtual function for inheritance.
+   * Use this function for cleaning tasks (e.g. destroying arrays, file outut)
+   *
+   **/
+  virtual void cleanup(){};
+
+private:
+  //! Specialized move to be used for the movement of the monomers 
+  MoveType move;
+  //! 
+  ConnectionMoveType connectionMove;
+  //! Number of mcs to be executed
+  uint32_t nsteps;
+  //! random number generator (seed set in main program)
+  RandomNumberGenerators rng;
+  //! 
+  IngredientsType& ingredients;
+  //! number of monomers which could have a reaction bond
+  uint32_t NReactiveSites;
+  //! number of already reacted monomers
+  uint32_t NReactedSites;
+  //! get conversion of the reaction process
+  double getConversion(){return (1.*NReactedSites)/(1.*NReactiveSites);};
+  
+};
+/**Implementation of the member functions
+ * @brief 
+ */
+template<class IngredientsType,class MoveType, class ConnectionMoveType>
+bool UpdaterSimpleConnection<IngredientsType,MoveType,ConnectionMoveType>::execute()
+{
+  
+      time_t startTimer = time(NULL); //in seconds
+      std::cout<<"connection mcs "<<ingredients.getMolecules().getAge() << " passed time " << ((difftime(time(NULL), startTimer)) ) <<std::endl;
+      for(int n=0;n<nsteps;n++)
+      {
+	  std::vector<uint32_t> TagedMonomers; 
+	for(size_t m=0;m<ingredients.getMolecules().size();m++)
+	{
+		if( ingredients.getMolecules()[m].IsReactive() ) TagedMonomers.push_back(m);
+		move.init(ingredients);
+		if(move.check(ingredients)==true)
+		{
+			move.apply(ingredients);
+		}
+	}
+	uint32_t nTags(TagedMonomers.size());
+	
+	for(uint32_t i =0; i < nTags;i++)
```

Comment:
This implicit increase the MCS. Maybe an alternative approach is useful? See latest commit.